chain/network/src/peer_manager/tests/nonce.rs

```diff
@@ -75,3 +78,57 @@ async fn test_nonces() {
         }
     }
 }
+
+async fn wait_for_edge(actor_handler: &mut ActorHandler) -> Edge {
+    actor_handler
+        .events
+        .recv_until(|ev| match ev {
+            Event::PeerManager(peer_manager_actor::Event::EdgesVerified(ev)) => Some(ev[0].clone()),
+            _ => None,
+        })
+        .await
+}
+
+#[tokio::test]
+/// Create 2 peer managers, that connect to each other.
+/// Verify that the will refresh their nonce after some time.
+async fn test_nonce_refresh() {
+    init_test_logger();
+    let mut rng = make_rng(921853255);
+    let rng = &mut rng;
+    let mut clock = time::FakeClock::new(*EDGE_MIN_TIMESTAMP_NONCE + time::Duration::days(2));
+    let chain = Arc::new(data::Chain::make(&mut clock, rng, 10));
+
+    // Start a PeerManager.
+    let pm = peer_manager::testonly::start(
+        clock.clock(),
+        near_store::db::TestDB::new(),
+        chain.make_config(rng),
+        chain.clone(),
+    )
+    .await;
+
+    let mut other_config = chain.make_config(rng);
+    // Make sure that other peer manager has the first one as the boot node, and that it is ready to connect to it.
+    other_config.peer_store.boot_nodes.push(pm.peer_info());
+    other_config.outbound_disabled = false;
```

Comment:
the preferred way is to use pm.connect_to() and default config (i.e. with outbound_disabled = true) so that no spontaneus connections are created (to make the test more deterministic). Unless you actually want to cover the peer selection logic here.

Reply:
nice. done.